apps/exports/views.py

```diff
@@ -0,0 +1,324 @@
+        meta = self.model._meta
+        exclude = self.exclude if self.exclude else []
+
+        if self.fields:
+            fields = [meta.get_field(name) for name in self.fields]
+        else:
+            fields = meta.get_fields()
+
+        names = ['link']
+        header = [_('Link')]
+        for field in fields:
+            if field.concrete \
+                    and not (field.one_to_one and field.rel.parent_link) \
+                    and field.name not in exclude \
+                    and field.name not in names:
+
+                names.append(field.name)
+                header.append(str(field.verbose_name))
+
+        virtual = self.get_virtual_fields()
+        for name, head in virtual.items():
+            if name not in names:
+                names.append(name)
+                header.append(head)
+
+        return header, names
+
+    def get_queryset(self):
+        return super().get_queryset().filter(module=self.module)
+
+    def get_virtual_fields(self):
+        virtual = super().get_virtual_fields()
+        if 'link' not in virtual:
+            virtual['link'] = _('Link')
+        return virtual
+
+    def get_header(self):
+        return self._header
+
+    def export_rows(self):
+        for item in self.get_queryset().all():
+            yield [self.get_field_data(item, name) for name in self._names]
+
+    def get_field_data(self, item, name):
+        # Use custom getters if they are defined
+        get_field_attr_name = 'get_%s_data' % name
+        if hasattr(self, get_field_attr_name):
+            get_field_attr = getattr(self, get_field_attr_name)
+
+            if hasattr(get_field_attr, '__call__'):
+                return get_field_attr(item)
+            return get_field_attr
+
+        # If item is a dict, return the fields data by key
+        try:
+            if name in item:
+                return item['name']
+        except TypeError:
+            pass
+
+        # Finally try to get the fields data as a property
+        return getattr(item, name, '')
+
+    def get_link_data(self, item):
+        return self.request.build_absolute_uri(item.get_absolute_url())
+
+    def get_description_data(self, item):
+        return strip_tags(item.description.strip())
+
+    def get_creator_data(self, item):
+        return item.creator.username
+
+    def get_created_data(self, item):
+        return item.created.isoformat()
+
+
+class ItemExportWithRatesMixin(VirtualFieldMixin):
+    def get_virtual_fields(self):
+        virtual = super().get_virtual_fields()
+        if 'ratings_positive' not in virtual:
+            virtual['ratings_positive'] = _('Positive ratings')
+        if 'ratings_negative' not in virtual:
+            virtual['ratings_negative'] = _('Negative ratings')
+
+        return virtual
+
+    def _count_ratings(self, item, value):
+        ct = ContentType.objects.get_for_model(item)
+        return Rating.objects.filter(content_type=ct, value=value).count()
```

Comment:
im trying to get the annotated rating in the get_ratings_x_data methods. the _count_ratings is only called if it is not available